pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -418,19 +398,20 @@ func TestHibernateAfter(t *testing.T) {
 			result, err := reconciler.Reconcile(reconcile.Request{
 				NamespacedName: types.NamespacedName{Namespace: namespace, Name: cdName},
 			})
-			if test.expectRequeueAfter {
-				assert.True(t, result.RequeueAfter > 0)
-			} else {
-				assert.Equal(t, 0*time.Second, result.RequeueAfter)
+
+			// Need to do fuzzy requeue after matching
+			if assert.Nil(t, err) {
+				assert.True(t, test.expectRequeueAfter-10*time.Second < result.RequeueAfter &&
```

Comment:
If `expectedRequeueAfter` is zero, you should check expliictly for zero without the 10-second buffer.